flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -228,55 +245,55 @@ public WebRuntimeMonitor(
 		} else {
 			serverSSLContext = null;
 		}
-		metricFetcher = new MetricFetcher(retriever, queryServiceRetriever, executor, timeout);
+		metricFetcher = new MetricFetcher(retriever, queryServiceRetriever, scheduledExecutor, timeout);
 
 		String defaultSavepointDir = config.getString(CoreOptions.SAVEPOINT_DIRECTORY);
 
-		JobCancellationWithSavepointHandlers cancelWithSavepoint = new JobCancellationWithSavepointHandlers(currentGraphs, executor, defaultSavepointDir);
+		JobCancellationWithSavepointHandlers cancelWithSavepoint = new JobCancellationWithSavepointHandlers(executionGraphCache, scheduledExecutor, defaultSavepointDir);
 		RuntimeMonitorHandler triggerHandler = handler(cancelWithSavepoint.getTriggerHandler());
 		RuntimeMonitorHandler inProgressHandler = handler(cancelWithSavepoint.getInProgressHandler());
 
 		Router router = new Router();
 		// config how to interact with this web server
-		get(router, new DashboardConfigHandler(executor, cfg.getRefreshInterval()));
+		get(router, new DashboardConfigHandler(scheduledExecutor, cfg.getRefreshInterval()));
```

Comment:
well i see an easy way to reduce the diff...

Reply:
But it's not necessarily about reducing the diff if the name change makes it clearer what this object is.